core/shard/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+extern crate chain;
+extern crate parking_lot;
+extern crate primitives;
+extern crate rand;
+#[macro_use]
+extern crate serde_derive;
+extern crate storage;
+
+use chain::{Block, Header};
+use primitives::hash::{CryptoHash, hash_struct};
+use primitives::types::{AuthorityMask, MerkleHash, MultiSignature, PartialSignature, ReceiptTransaction, SignedTransaction};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ShardBlockHeaderBody {
+    pub parent_hash: CryptoHash,
+    pub index: u64,
+    pub merkle_root_state: MerkleHash,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ShardBlockHeader {
+    pub body: ShardBlockHeaderBody,
+    pub hash: CryptoHash,
+    pub authority_mask: AuthorityMask,
+    pub signature: MultiSignature,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ShardBlockBody {
```

Comment:
Both Beacon and Shard blocks should be composed of `SignedMessageData` because we need it for slashing and catch up. Transactions and Receipts can be extracted from the `SignedMessageData`. Also, they should not be ordered separately, so we cannot keep them in separate vectors.

Reply:
We should do transactions and receipts together and store result of consensus from TxFlow.
Filed as separate issue in #174